src/Domain/CampaignRepository.php

```diff
@@ -14,6 +14,23 @@
  */
 class CampaignRepository extends SalesforceReadProxyRepository
 {
+    public function getCampaignBySalesforceId(string $salesforceId, bool $withResultCache): ?Campaign
```

Comment:
probably for simplicity we could set 
```suggestion
    public function getCampaignBySalesforceId(string $salesforceId, bool $withResultCache = true): ?Campaign
```

and only pass the argument when it's false. Otherwise it's not obvious unless you know Doctrine better than I do that using the result cache is the default, and we only disable it an edge case semi-unhappy path. 

Same applies to the param of `\MatchBot\Domain\DonationRepository::buildFromApiRequest` which is only called twice from prod code, with false just after the `UniqueConstraintViolationException` is caught.